File: steps/processor.step.ts
import type { Handlers } from 'motia'
 
export const handler: Handlers['Processor'] = async (req, { state, traceId }) => {
  // Set a value
  await state.set(traceId, 'userCount', 42)
 
  // Get a value
  const count = await state.get<number>(traceId, 'userCount')
 
  // Get all keys for a trace
  const keys = await state.keys(traceId)
 
  // Delete a value
  await state.delete(traceId, 'userCount')
 
  // Clear all state for a trace
  await state.clear(traceId)
 
  return { status: 200, body: { count } }
}

File: steps/processor_step.py
async def handler(req, context):
    # Set a value
    await context.state.set(context.trace_id, "userCount", 42)
 
    # Get a value
    count = await context.state.get(context.trace_id, "userCount")
 
    # Get all keys for a trace
    keys = await context.state.keys(context.trace_id)
 
    # Delete a value
    await context.state.delete(context.trace_id, "userCount")
 
    # Clear all state for a trace
    await context.state.clear(context.trace_id)
 
    return {"status": 200, "body": {"count": count}}